railgun/manager/prometheus.go

```diff
@@ -0,0 +1,102 @@
+package manager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/ctrlutils"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/mgrutils"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func RunHTTP(log logr.Logger) {
+	stopCh := make(chan struct{})
+	var wg sync.WaitGroup
+	mux := http.NewServeMux()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ServeHTTP(true,
+			ctrlutils.PROMTHPORT, mux, stopCh,
+			log,
+			&wg)
+	}()
+}
+
+// ServeHTTP enable HTTP Server
+// - prometheus
+func ServeHTTP(enableProfiling bool,
+	port int,
+	mux *http.ServeMux,
+	stop <-chan struct{},
+	logger logr.Logger,
+	wg *sync.WaitGroup) {
+	defer wg.Done()
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	mux.Handle("/metrics", promhttp.Handler())
+
+	mux.HandleFunc("/build", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		b, _ := json.Marshal(mgrutils.Version("UNKNOWN", "UNKNOWN", mgrutils.KICREPO))
+		if _, err := w.Write(b); err != nil {
+			logger.Error(err, " endpoint /build failed to write response.")
+		}
+	})
+
+	mux.HandleFunc("/stop", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
why do we need this? If there is no demonstrated need, I suggest dropping this:
- because YAGNI
- for security reasons: this allows everybody with raw HTTP access to KIC to perform a DoS attack by killing the controller.